flight/Modules/Logging/logging.c

```diff
@@ -232,107 +219,92 @@ static void loggingTask(void *parameters)
+				// log all UAVObjects, settings and data alike.
+				LL_FOREACH(log_info, info) {
+					// Iterate over all instances.
+					uint16_t numInstances = UAVObjGetNumInstances(info->obj);
+					for (int instId=0; instId<numInstances; instId++) {
+						UAVTalkSendObjectTimestamped(uavTalkCon, info->obj, instId, false, 0);
 					}
+					// use a random offset, so that not all objects are logged at the same time
+					info->next_log =  PIOS_Thread_Systime() + randomize_int(info->logging_period);
 				}
-
-				// Trigger logging for objects that are logged on change
-				flightstatus_updated = true;
-				waypoint_updated = true;
-
-				first_run = false;
-			}
-
-			// Log objects on change
-			if (flightstatus_updated){
-				UAVTalkSendObjectTimestamped(uavTalkCon, FlightStatusHandle(), 0, false, 0);
-				flightstatus_updated = false;
-			}
-
-			if (waypoint_updated && WaypointActiveHandle()){
-				UAVTalkSendObjectTimestamped(uavTalkCon, WaypointActiveHandle(), 0, false, 0);
-				waypoint_updated = false;
 			}
-
-			// Log very fast
-			UAVTalkSendObjectTimestamped(uavTalkCon, AccelsHandle(), 0, false, 0);
-			UAVTalkSendObjectTimestamped(uavTalkCon, GyrosHandle(), 0, false, 0);
-
-			// Log a bit slower
-			if ((i % 2) == 0) {
-				UAVTalkSendObjectTimestamped(uavTalkCon, AttitudeActualHandle(), 0, false, 0);
-				UAVTalkSendObjectTimestamped(uavTalkCon, MagnetometerHandle(), 0, false, 0);
-				UAVTalkSendObjectTimestamped(uavTalkCon, ManualControlCommandHandle(), 0, false, 0);
-				UAVTalkSendObjectTimestamped(uavTalkCon, ActuatorCommandHandle(), 0, false, 0);
-			}
-
-			// Log slower
-			if ((i % 10) == 1) {
-				UAVTalkSendObjectTimestamped(uavTalkCon, BaroAltitudeHandle(), 0, false, 0);
-				if (AirspeedActualHandle())
-					UAVTalkSendObjectTimestamped(uavTalkCon, AirspeedActualHandle(), 0, false, 0);
-				if (GPSPositionHandle())
-					UAVTalkSendObjectTimestamped(uavTalkCon, GPSPositionHandle(), 0, false, 0);
-				if (PositionActualHandle())
-					UAVTalkSendObjectTimestamped(uavTalkCon, PositionActualHandle(), 0, false, 0);
-				if (VelocityActualHandle())
-					UAVTalkSendObjectTimestamped(uavTalkCon, VelocityActualHandle(), 0, false, 0);
+			// Empty the queues
+			while(PIOS_Queue_Receive(shared_queue, &ev, 0))
+			LL_FOREACH(log_info, info) {
+				if (info->queue == shared_queue){
+					// The list is sorted based on the logging period, so we can stop as soon
+					// as we have an item without a private queue (as subsequent items won't 
+					// have a private queue either).
+					break;
+				}
+				while(PIOS_Queue_Receive(info->queue, &ev, 0)){};
 			}
+			LoggingStatsBytesLoggedSet(&written_bytes);
+			loggingData.Operation = LOGGINGSTATS_OPERATION_LOGGING;
+			LoggingStatsSet(&loggingData);
+			break;
+		case LOGGINGSTATS_OPERATION_LOGGING:
+			{
+				// Sleep between writing
+				PIOS_Thread_Sleep_Until(&now, LOGGING_PERIOD_MS);
+
+				// Log the objects with private queues
+				for (int i=0; i<PRIVATE_QUEUE_SIZE; i++){
```

Comment:
> The only concern is the increased memory consumption, i.e., I'm not sure if the F1 targets could handle it if it is implemented in the way that every callback has an interval associated with it. 

You don't like my proposed solution to that in https://github.com/TauLabs/TauLabs/pull/1928#discussion_r41418120  ?  If you do that, there's no need for parallel data structures or additional lookups and only callbacks with interval pay a 4 or 8 byte penalty on size.

I can mock it up / do an initial implementation if unclear.